### PR TITLE
feat: allow keyboard shortcut creation for terminal copy commands

### LINKED ISSUE #197098: allow keyboard shortcut creation for 1) copying last command and 2) copying last command **and** its output
There should be a way to add keyboard shortcuts for
1. copying the last command
2. copying the last command's output (this already exists), and
3. copying the last command **and** its output.

### PR DESCRIPTION
**Pull request by hirehamir**
fixes #197098.

### CODE REVIEW

**Comment by hamirmahal** [comment]
# How to test
1. Run `yarn watch` in one terminal.
2. Run `./scripts/code.sh` in another terminal.
3. Open keyboard shortcuts in the `Code - OSS Dev` window, and note that it is now possible to add shortcuts for copying just the last command, and copying the last command **with** its output, in addition to the preexisting shortcut to just copy the last command's output.

**Comment by hamirmahal** [comment]
# Without this change
file:///home/hamir/Pictures/Screenshots/Screenshot%20from%202023-10-31%2010-53-57.png

**Comment by hamirmahal** [comment]
# Without this change
Screenshot from 2023-10-31 10-53-57

**Comment by hamirmahal** [comment]
# Without this change
image

**Comment by hamirmahal** [comment]
# With this change
[Screencast from 10-31-2023 12:51:27 PM.webm](https://github.com/microsoft/vscode/assets/43425812/f018fd50-13cf-44cf-b630-c48d6ed24b7b)

File: src/vs/workbench/contrib/terminal/browser/terminalActions.ts [hunk 1]
```diff
@@ -454,6 +472,28 @@ export function registerTerminalActions() {
...
+			if (!commands || commands.length === 0) {
+				return;
+			}
+			const command = commands[commands.length - 1];
+			if (!command?.hasOutput()) {
+				return;
+			}
+			const output = command.getOutput();
+			if (isString(output)) {
+				await clipboardService.writeText(`${command.command !== '' ? command.command + '\n' : ''}${output}`);
```
Comment (hamirmahal) [inline src/vs/workbench/contrib/terminal/browser/terminalActions.ts:491]: I copied this logic from https://github.com/microsoft/vscode/pull/192217/files#diff-f26cf22b79d035fac45646c72f194cc92cf9629bd5cb58e8d8a39d220bb0d5dcR389.

Comment (hamirmahal) [reply src/vs/workbench/contrib/terminal/browser/terminalActions.ts:491]: Maybe it's worth extracting this duplicated logic to a utility somewhere?

**Comment by nangmangorani** [comment]
여기다추가

**Comment by nangmangorani** [comment]
> 여기다추가

File: src/vs/workbench/contrib/terminal/browser/terminalActions.ts [hunk 2]
```diff
@@ -4,67 +4,67 @@
...
 import { IRemoteTerminalAttachTarget, ITerminalConfigHelper, ITerminalProfileResolverService, ITerminalProfileService, TerminalCommandId } from 'vs/workbench/contrib/terminal/common/terminal';
 import { TerminalContextKeys } from 'vs/workbench/contrib/terminal/common/terminalContextKey';
-import { createProfileSchemaEnums } from 'vs/platform/terminal/common/terminalProfiles';
 import { terminalStrings } from 'vs/workbench/contrib/terminal/common/terminalStrings';
 import { IConfigurationResolverService } from 'vs/workbench/services/configurationResolver/common/configurationResolver';
+import { AbstractVariableResolverService } from 'vs/workbench/services/configurationResolver/common/variableResolver';
+import { editorGroupToColumn } from 'vs/workbench/services/editor/common/editorGroupColumn';
+import { IEditorGroupsService } from 'vs/workbench/services/editor/common/editorGroupsService';
+import { SIDE_GROUP } from 'vs/workbench/services/editor/common/editorService';
 import { IWorkbenchEnvironmentService } from 'vs/workbench/services/environment/common/environmentService';
```
Comment (meganrogge) [inline src/vs/workbench/contrib/terminal/browser/terminalActions.ts:64]: Most of these import changes should be reverted

Comment (hamirmahal) [reply src/vs/workbench/contrib/terminal/browser/terminalActions.ts:64]: Okay, I'll take a look now.

Comment (hamirmahal) [reply src/vs/workbench/contrib/terminal/browser/terminalActions.ts]: Do you think it's worth updating `.vscode/settings.json` to have this, either in this pull request, or a separate one?

```json
	"[typescript]": {
		"editor.defaultFormatter": "vscode.typescript-language-features",
		"editor.formatOnSave": true,
		"editor.codeActionsOnSave": { // new
			"source.organizeImports": false // new
		}
	},
```

Comment (hamirmahal) [reply src/vs/workbench/contrib/terminal/browser/terminalActions.ts]: Done

Comment (hamirmahal) [reply src/vs/workbench/contrib/terminal/browser/terminalActions.ts]: Do you think it's worth updating `.vscode/settings.json` to have this, either in this pull request, or a separate one?

```json
	"[typescript]": {
		"editor.defaultFormatter": "vscode.typescript-language-features",
		"editor.formatOnSave": true,
		"editor.codeActionsOnSave": { // new
			"source.organizeImports": false // new
		}
	},
```

Comment (meganrogge) [reply src/vs/workbench/contrib/terminal/browser/terminalActions.ts]: I believe that is false by default, no?

Comment (hamirmahal) [reply src/vs/workbench/contrib/terminal/browser/terminalActions.ts]: Ah, I think you're right.

**Review by meganrogge — APPROVED** [review]
Thanks, this works well. 

@Tyriar I'm wondering if we should have `f1:false` to hide these from the command palette?

**Comment by Tyriar** [comment]
@meganrogge they're fine to be in the command palette, the main criteria for f1:false imo is if it's duplicating things, internal or wouldn't make sense for a user to do or discover there.

**Review by Tyriar — APPROVED** [review]

**Comment by Tyriar** [comment]
@0x303 it should be out in the latest stable version as of a few days ago

**Pull request merged**